tools/src/test/kotlin/ktx/tools/BundleLinesCreatorTest.kt

```diff
@@ -0,0 +1,210 @@
+  }
+
+  @Test
+  fun `should create valid enum entry names`() {
+    val bundleLinesCreator = object : BundleLinesCreator() {
+      public override fun convertToEntryName(name: String) = super.convertToEntryName(name)
+    }
+    val input = listOf("key", "%&weird\\u0020key", "怪异的钥匙", "", " ", "\t", "backtick`key")
+    val output = input.mapNotNull(bundleLinesCreator::convertToEntryName)
+    assertTrue(output.all(String::isNotEmpty))
+    assertTrue(output.all { string ->
+      (!string.first().isDigit() && string.all { it.isLetterOrDigit() || it == '_' }) ||
+        (string.first() == '`' && string.last() == '`' && string.count { it == '`' } == 2)
+    })
+  }
+
+  @Test
+  fun `should produce expected code`() {
+    val bundleLinesCreator = object : BundleLinesCreator() {
+      public override fun generateKtFileContent(
+        packageName: String,
+        enumClassName: String,
+        entryNames: Set<String>,
+        indent: String
+      ): String {
+        return super.generateKtFileContent(packageName, enumClassName, entryNames, indent)
+      }
+    }
+    val packageName = "com.packagename"
+    val className = "Nls"
+    val indent = "\t"
+    val entryNames = setOf("key", "secondKey", "`%&weird key`")
+    val code = bundleLinesCreator.generateKtFileContent(packageName, className, entryNames, indent)
+
+    val packageDeclaration = Regex("""^package\s.*""").find(code)?.value?.trim()
+    assertEquals("package $packageName", packageDeclaration)
+
+    val importDeclarations = Regex("""\nimport\s.*""").findAll(code)
+      .map { it.value.trim() }
+      .toList()
+    val requiredDeclarations = setOf(
+      "import ktx.i18n.BundleLine",
+      "import com.badlogic.gdx.utils.I18NBundle"
+    )
+    assertTrue(
+      "Must include proper BundleLine and I18NBundle imports, but the imports were $importDeclarations.",
+      importDeclarations.containsAll(requiredDeclarations)
+    )
+
+    val enumDeclaration = Regex("""enum\s.*(?=\{)""").find(code)?.value?.trim()
+      ?.replace(Regex("""\b\s*:\s*\b"""), ":") // trim unnecessary inner whitespace
+    assertEquals("enum class $className:BundleLine", enumDeclaration)
+
+    val generatedEntryNames = Regex("""(?<=\{)[\s\S]*(?=;)""").find(code)
+      ?.value?.trim()
+      ?.split(Regex("""\s*,\s*"""))
+      ?.toSet()
+    assertEquals(entryNames, generatedEntryNames)
+
+    fun checkBrackets(open: Char, close: Char): Boolean {
+      var openCount = 0
+      for (c in code) when (c) {
+        open -> ++openCount
+        close -> if (--openCount < 0) return false
+      }
+      return openCount == 0
+    }
+
+    assertTrue(
+      "Improperly matched parentheses in generated code:\n$code",
+      checkBrackets('(', ')')
+    )
+    assertTrue(
+      "Improperly matched curly braces in generated code:\n$code",
+      checkBrackets('{', '}')
+    )
+    assertTrue(
+      "Improperly matched square brackets in generated code:\n$code",
+      checkBrackets('[', ']')
+    )
```

Comment:
I think it makes sense to just compare it with an equality check to a multiline string with the expected output. Checking the matching brackets says nothing about the validity of the code.

Reply:
OK, replace the entire test with a simple equality check against a single hard-coded String? I wasn't sure about that because there are changes to the output we could make that wouldn't be errors, but I suppose we could just update the test accordingly if we did that.

Reply:
If the output changes and the tests do not catch that, I'd argue that the tests are incomplete. We're supposed to ensure reproducibility and stability of the generated code.